src/components/date-range-picker.ts

```diff
@@ -0,0 +1,221 @@
+            color: var(--primary-text-color);
+          }
+          .daterangepicker td.active,
+          .daterangepicker td.active:hover {
+            background-color: var(--primary-color);
+            color: var(--text-primary-color);
+          }
+          .daterangepicker td.start-date.end-date {
+            border-radius: 50%;
+          }
+          .daterangepicker td.start-date {
+            border-radius: 50% 0 0 50%;
+          }
+          .daterangepicker td.end-date {
+            border-radius: 0 50% 50% 0;
+          }
+          .reportrange-text {
+            background: none !important;
+            padding: 0 !important;
+            border: none !important;
+          }
+          .daterangepicker .calendar-table .next span,
+          .daterangepicker .calendar-table .prev span {
+            border: solid var(--primary-text-color);
+            border-width: 0 2px 2px 0;
+          }
+          .daterangepicker .ranges li {
+            outline: none;
+          }
+          .daterangepicker .ranges li:hover {
+            background-color: var(--secondary-background-color);
+          }
+          .daterangepicker .ranges li.active {
+            background-color: var(--primary-color);
+            color: var(--text-primary-color);
+          }
+          .daterangepicker select.ampmselect,
+          .daterangepicker select.hourselect,
+          .daterangepicker select.minuteselect,
+          .daterangepicker select.secondselect {
+            background: transparent;
+            border: 1px solid var(--divider-color);
+            color: var(--primary-color);
+          }
+          .daterangepicker .drp-buttons .btn {
+            border: 1px solid var(--primary-color);
+            background-color: transparent;
+            color: var(--primary-color);
+            border-radius: 4px;
+            padding: 8px;
+            cursor: pointer;
+          }
+          .calendars-container {
+            flex-direction: column;
+            align-items: center;
+          }
+          .drp-calendar.col.right .calendar-table {
+            display: none;
+          }
+          .daterangepicker.show-ranges .drp-calendar.left {
+            border-left: 0px;
+          }
+          .daterangepicker .drp-calendar.left {
+            padding: 8px;
+          }
+          .daterangepicker.show-calendar .ranges {
+            margin-top: 0;
+            padding-top: 8px;
+            border-right: 1px solid var(--divider-color);
+          }
+          @media only screen and (max-width: 800px) {
+            .calendars {
+              flex-direction: column;
+            }
+          }
+          .calendar-table {
+            padding: 0 !important;
+          }     
+        `;
+    const shadowRoot = this.shadowRoot!;
+    shadowRoot.appendChild(style);
+    shadowRoot.addEventListener("click", (ev) => ev.stopPropagation());
+  }
+}
+
+window.customElements.define("date-range-picker", DateRangePickerElement);
```

Comment:
Why not use the `@customElement` decorator?

Also don't forget to add to HTML map for TS.